### PR TITLE
add comment about half

### LINKED ISSUE #867: HalfTensor support
Got `RuntimeError: torch_xla/csrc/tensor_util.cpp:652 : Type not supported: Half` error when I try to convert HalfTensor to xla. If we decide not to support it for now, we should probably add it to our 2 pages guidelines.

```
import torch
import torch_xla

device = 'xla:0'

a = torch.rand(3, 3).half()
b = a.to(device)
print(b)
```

### PR DESCRIPTION
**Pull request by ailzhang**
fixes #867

### CODE REVIEW

File: API_GUIDE.md [hunk 1]
```diff
@@ -16,6 +16,8 @@ print(x)
 
 The XLA device is not a physical device but instead stands in for either a Cloud TPU or CPU. The underlying storage for XLA tensors is a contiguous buffer in device memory and the code in the model shouldn't assume any stride.
 
+XLA Tensor doesn't support converting single tensor to half precision using `tensor.half()`, instead environment variable `XLA_USE_BF16` is available to tranforms **all** the PyTorch Float values into BiFloat16 when sending to the TPU device.
```
Comment (asuhan) [inline API_GUIDE.md]: Make this two sentences: "... using `tensor.half()`. Instead, environment variable `XLA_USE_BF16`...".

Comment (asuhan) [reply API_GUIDE.md]: The format is called `bfloat16`, not `BiFloat16`.

Comment (asuhan) [reply API_GUIDE.md]: "... which converts **all** PyTorch float values to bfloat16 when sending them to the TPU device."

File: API_GUIDE.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ print(x)
 
 The XLA device is not a physical device but instead stands in for either a Cloud TPU or CPU. The underlying storage for XLA tensors is a contiguous buffer in device memory and the code in the model shouldn't assume any stride.
 
-XLA Tensor doesn't support converting single tensor to half precision using `tensor.half()`, instead environment variable `XLA_USE_BF16` is available to tranforms **all** the PyTorch Float values into BiFloat16 when sending to the TPU device.
+XLA Tensor doesn't support converting single tensor to half precision using `tensor.half()`. Instead, environment variable `XLA_USE_BF16` is available, which converts **all** PyTorch float values to bfloat16 when sending them to the TPU device.
```
Comment (dlibenzi) [inline API_GUIDE.md]: I'd add something like:
```The conversion is  totally transparent to the user, and the XLA tensors will still retain a float dtype. Similarly, when the tensor is moved back to CPU, it's type will be float.```

**Comment by ailzhang** [comment]
cc: @dlibenzi any more suggestions here? ;)

**Pull request merged**